openvalidation-generation/src/main/resources/java/framework.hbs

```diff
@@ -390,12 +390,16 @@ public List<ValidationRule> Rules = new ArrayList<>();
         return !AT_LEAST_ONE_OF(leftOperand, rightOperand);
     }
 
-    public <T extends Number> Double SUM_OF(T... items)
+    public <T extends Number> BigDecimal SUM_OF(T... items)
     {
         if (items == null || items.length < 1)
-            return 0.0;
+            return new BigDecimal(0);
 
-        return Arrays.stream(items).mapToDouble(i -> i.doubleValue()).sum();
+        BigDecimal value = new BigDecimal(0);
+        for(T item : items){
+            value = BigDecimal.valueOf(item.doubleValue()).add(value);
+        }
+        return value;
```

Comment:
... cast `value` to `T` here?

Or even not use BigDecimal at all. Hmm.
Imagine the user used a custom implementation of `Number` in the model, that overloads the add operator?
Maybe I'm overthinking it with the generality here.
This is probably fine.